client.go

```diff
@@ -0,0 +1,56 @@
+package compute
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+type Client struct {
+	config     *Config
+	httpClient *http.Client
+}
+
+func NewClient(config *Config, httpClient *http.Client) (*Client, error) {
+	if config == nil {
+		return nil, errors.New("failed to NewClient: config cannot be nil")
+	}
+
+	if httpClient == nil {
+		return nil, errors.New("failed to NewClient: httpClient cannot be nil")
+	}
+
+	client := &Client{
+		config:     config,
+		httpClient: httpClient,
+	}
+
+	return client, nil
+}
+
+func NewLocalClient() (*Client, error) {
```

Comment:
I had a bit of a brain fart when reading this—maybe add a comment here saying that it uses URIs for my local installation?